.github/workflows/publish.yml

```diff
@@ -37,59 +37,22 @@ jobs:
   publish-or-pr:
     if: github.event_name == 'push'
     permissions:
-      contents: write # changesets/action
+      contents: write
       issues: write
       pull-requests: write
-      id-token: write # OIDC for provenance if npm publish happens here
+      id-token: write
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
+      - name: Setup Project
```

Comment:
There should be nothing different, i just refactored this to be more "clean" and use a composite action that can be reused across steps.